config/initializers/rack_attack.rb

```diff
@@ -38,6 +38,8 @@ class Rack::Attack
   def self.protected_route?(protected_actions, path, method)
     route_params = Rails.application.routes.recognize_path(path, method: method)
     protected_actions.any? { |hash| hash[:controller] == route_params[:controller] && hash[:action] == route_params[:action] }
+  rescue ActionController::RoutingError # raised for routes with a redirect
+    false
```

Comment:
I keep getting suspicious of this `rescue`. Feels ripe for abuse to bypass the rate limits.
There was plan of [`recognize_path` supporting `Redirect`](https://github.com/rails/rails/pull/6922/files#diff-ccdef70ce0e95afc8816f47f89b56c60R657) but it didn't get through, let us their fix.
At very least, we should raise the error back if `Rails.application.routes.call(req.env)[0]` is not `301`.

@dwradcliffe any opinion on this? It was your idea to support this. We could be using fastly or nginx, which already have other redirects we use.